model/team.go

```diff
@@ -34,11 +32,12 @@ type Team struct {
 	Name               string  `json:"name"`
 	Description        string  `json:"description"`
 	Email              string  `json:"email"`
-	Type               string  `json:"type"`
+	Type               *string `json:"type"` // Do not use, this will be deprecated in favor of InviteId == '' or InviteId != ''
 	CompanyName        string  `json:"company_name"`
 	AllowedDomains     string  `json:"allowed_domains"`
 	InviteId           string  `json:"invite_id"`
-	AllowOpenInvite    bool    `json:"allow_open_invite"`
+	IsPublic           *bool   `json:"is_public"`
+	AllowOpenInvite    *bool   `json:"allow_open_invite"` // Do not use, this will be deprecated in favor of IsPublic
```

Comment:
In the past (with the `config`) we've renamed these struct fields to `DEPRECATED_DO_NOT_USE_AllowOpenInvite` so that people don't let more use of them slip in through subsequent PRs. Do you think we should consider doing that here (even though it breaks source compatibility) or should we not?

Reply:
Actually, thinking about it more, I don't think we need to do that, because here those fields are still safe to use, whereas in the config case, their values were actually incorrect.

Reply:
Well, my idea is to deal with the compatibility in the borders of the application, API, Plugins API and Import/Export, internally, the data will be consistent and we should use "IsPublic" only. If you are messing with this fields using the internal API, is your problem.